### PR TITLE
client track page

### PR DESCRIPTION
**Pull request by j-abellera**
Created client-track page for clients to fill out a form to view their scheduled appointment.
Form uses yup for schema, validation, and error handling.
Created route and action for GET requests when the form is submitted, used mock data to flesh things out.
Also created a details nested page if the GET request finds a matching appointment to display.
route and action created to handle feature for cancelling appointment. 
Added Context API to handle state between components.
Added general styling for the meantime.

### CODE REVIEW

File: client/components/client-track/clientTrack.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import './client-track.css';
+import { useRouter } from 'next/navigation';
+import { getAppointment } from '../../actions/client-track';
+import { useClientContext } from '../../context/clientContext';
+
+//Form validation using yup
+import * as yup from 'yup';
```
Comment (tdkent) [inline client/components/client-track/clientTrack.js:10]: Not a big deal, but we have already added Joi for validation. If yup is preferred we can easily modify, but we should probably use one validation pkg.

**Review by tdkent — COMMENTED** [review]
Should be fine to merge for now. I do have a few notes:

- I may be unclear on what the UX is meant to be in having both an appt tracking and appt details page? Could these pages possibly be combined?
- Is it worth wrapping the entire app with an appt info context? Unless I'm missing something, is this to allow a redirect to the `/details` page after the fetch request?
- CSS modules can be updated to Tailwind
- Plain HTML should be updated to MUI components for consistency
- API methods can be moved from `routes` to `controllers` for clearer architecture
- For consistency in local testing we should put the url to the server in an env variable so we do not have to manually change port numbers (not an issue with this PR, just something to keep in mind).

File: client/actions/client-track.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use server'
+
+import axios from'axios';
```
Comment (DalySoftware) [inline client/actions/client-track.js:3]: I'm not sure there's any use for axios these days? I think your code would look pretty much identical with fetch

File: client/actions/client-track.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            method: 'GET',
+            headers: {"Content-Type": "application/json"},
+            params: {email: formEmail, phone: formPhone}
+        });
+        console.log('returns', data);
+        console.log('from action', data.data.data);
+        return data.data.data;
+    } catch(err) {
+        console.log(err.response.data);
+        return err.response.data;
```
Comment (DalySoftware) [inline client/actions/client-track.js:19]: As a consumer, I don't think I'd expect to get an error data if I called `getAppointment()` and it didn't throw. I'd suggest this should either throw in an error state, or we should use a result pattern. Something like this (typescript syntax is just so I can illustrate)
```ts
{
  appointment: Appointment | undefined
  error: unknown | undefined
}
```

File: client/actions/client-track.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    try {
+        const res = await axios({
+            url: `http://localhost:9000/client-track/details/${id}`,
+            method: 'DELETE',
+            headers: {"Content-Type": "application/json"},
+        })
+
+        if(res.status === 204) {
+            console.log('Delete Successful');
+            return res.status;
```
Comment (DalySoftware) [inline client/actions/client-track.js:34]: Similar concern about polymorphic returns. I don't think it's intuitive that this returns either a numeric code or a boolean.

File: client/components/client-track/client-track.css [hunk 5]
```diff
@@ -0,0 +1,91 @@
+.client-track-container {
```
Comment (DalySoftware) [inline client/components/client-track/client-track.css:1]: In addition to what Tim says, if we're using MUI you don't often write raw css. Most of your styling should be either in the theme or using a styled() component. For one-off styles, the sx prop is normally used.

Good writeup here:
https://mui.com/material-ui/customization/how-to-customize/

**Review by DalySoftware — COMMENTED** [review]
> Is it worth wrapping the entire app with an appt info context? Unless I'm missing something, is this to allow a redirect to the /details page after the fetch request?

If we use SWR, react query or something, I think this problem goes away anyway. Each can just use the hook and rely on the query client caching.

**Comment by DalySoftware** [comment]
Can be reopened or re-raised later

**Pull request closed without merging**